aiida/manage/configuration/schema/config-v6.schema.json

```diff
@@ -1,358 +1,330 @@
+          "minimum": 1,
+          "description": "Timeout in seconds for communications with RabbitMQ"
+        },
+        "caching.default_enabled": {
+          "type": "boolean",
+          "default": false,
+          "description": "Enable calculation caching by default"
+        },
+        "caching.enabled_for": {
+          "description": "Calculation entry points to enable caching on",
+          "type": "array",
+          "default": [],
+          "items": {
+            "type": "string"
+          }
+        },
+        "caching.disabled_for": {
+          "description": "Calculation entry points to disable caching on",
+          "type": "array",
+          "default": [],
+          "items": {
+            "type": "string"
+          }
+        },
+        "autofill.user.email": {
+          "type": "string",
+          "global_only": true,
+          "description": "Default user email to use when creating new profiles."
+        },
+        "autofill.user.first_name": {
+          "type": "string",
+          "global_only": true,
+          "description": "Default user first name to use when creating new profiles."
+        },
+        "autofill.user.last_name": {
+          "type": "string",
+          "global_only": true,
+          "description": "Default user last name to use when creating new profiles."
+        },
+        "autofill.user.institution": {
+          "type": "string",
+          "global_only": true,
+          "description": "Default user institution to use when creating new profiles."
         }
+      }
     },
-    "required": [],
-    "properties": {
-        "CONFIG_VERSION": {
-            "description": "The configuration version",
-            "type": "object",
-            "required": [
-                "CURRENT",
-                "OLDEST_COMPATIBLE"
-            ],
-            "properties": {
-                "CURRENT": {
-                    "description": "Version number of configuration file format",
-                    "type": "integer",
-                    "const": 6
-                },
-                "OLDEST_COMPATIBLE": {
-                    "description": "Version number of oldest configuration file format this file is compatible with",
-                    "type": "integer",
-                    "const": 6
-                }
+    "profile": {
+      "type": "object",
+      "required": ["storage_backend", "storage_config", "process_control_backend", "process_control_config"],
+      "properties": {
+        "PROFILE_UUID": {
+          "description": "The profile's unique key",
+          "type": "string"
+        },
+        "storage_backend": {
+          "description": "The storage backend type to use",
+          "type": "string",
+          "default": "django"
+        },
+        "storage_config": {
+          "description": "The configuration to parse to the storage backend",
+          "type": "object",
+          "properties": {
+            "database_engine": {
```

Comment:
Although it is nice to have the explicit config options for the storage and process_control backends, this is problematic in principle since the config schema will be backend specific and so dynamic. Not sure if the JSON schema allows for such a concept. For now it is ok to keep since anyway we don't have a way to dynamically change the backends, but with your idea of making that possible in the future, just wanted to highlight this.

Reply:
Yeh, I was well aware of this incongruity, lol. Probably want to put this validation onto a class method on the storage/process_control backend, e.g.

```python
storage_cls = load_storage_cls(config["storage_backend"])
storage_cls.validate_config(config["storage_config"])
```